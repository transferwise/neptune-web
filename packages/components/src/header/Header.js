import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';

const Header = React.forwardRef((props, ref) => {
  const { bottomContent, className, layout, leftContent, rightContent } = props;
  const isVertical = layout === Header.Layout.VERTICAL;

  return (
    <div className={classNames('np-header', 'd-flex', 'flex-wrap', className)} ref={ref}>
      <div
        className={classNames('align-items-center', 'd-flex', {
          'flex__item--8': isVertical,
        })}
      >
        {leftContent}
      </div>

      <div
        className={classNames('align-items-center', 'd-flex', 'justify-content-end', {
          'flex__item--4 ': isVertical,
          'order-2': !isVertical,
        })}
      >
        {rightContent}
      </div>
      <div
        className={classNames('align-items-center', 'd-flex', 'justify-content-center', {
          'flex__item--12': isVertical,
          'order-1 flex-grow-1': !isVertical,
        })}
      >
        {bottomContent}
      </div>
    </div>
  );
});

Header.Layout = { VERTICAL: 'VERTICAL', HORIZONTAL: 'HORIZONTAL' };

Header.defaultProps = {
  bottomContent: undefined,
  className: undefined,
  layout: Header.Layout.HORIZONTAL,
  leftContent: undefined,
  rightContent: undefined,
};

Header.propTypes = {
  bottomContent: Types.node,
  className: Types.string,
  layout: Types.oneOf([Header.Layout.HORIZONTAL, Header.Layout.VERTICAL]),
  leftContent: Types.node,
  rightContent: Types.node,
};

export default Header;
