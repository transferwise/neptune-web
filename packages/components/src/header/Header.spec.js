import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import Header from './Header';

describe('Header', () => {
  const props = {
    leftContent: 'leftContent',
    rightContent: 'rightContent',
    bottomContent: 'bottomContent',
    className: 'className',
  };
  it('renders as expected', () => {
    expect(render(<Header {...props} />).container).toMatchSnapshot();
  });

  it('renders as expected with vertical layout', () => {
    expect(
      render(<Header {...props} layout={Header.Layout.VERTICAL} />).container,
    ).toMatchSnapshot();
  });
});
