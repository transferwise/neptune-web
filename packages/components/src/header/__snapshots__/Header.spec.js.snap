// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<div>
  <div
    class="np-header d-flex flex-wrap className"
  >
    <div
      class="align-items-center d-flex"
    >
      leftContent
    </div>
    <div
      class="align-items-center d-flex justify-content-end order-2"
    >
      rightContent
    </div>
    <div
      class="align-items-center d-flex justify-content-center order-1 flex-grow-1"
    >
      bottomContent
    </div>
  </div>
</div>
`;

exports[`Header renders as expected with vertical layout 1`] = `
<div>
  <div
    class="np-header d-flex flex-wrap className"
  >
    <div
      class="align-items-center d-flex flex__item--8"
    >
      leftContent
    </div>
    <div
      class="align-items-center d-flex justify-content-end flex__item--4 "
    >
      rightContent
    </div>
    <div
      class="align-items-center d-flex justify-content-center flex__item--12"
    >
      bottomContent
    </div>
  </div>
</div>
`;
